src/aioquic/h3/connection.py

```diff
@@ -230,6 +263,20 @@ def validate_headers(
         else:
             # regular headers
             after_pseudo_headers = True
+            # a few more semantic checks
+            if key == b"content-length":
+                try:
+                    content_length = int(value)
+                    if content_length < 0:
+                        raise ValueError
+                except ValueError:
+                    raise MessageError("content-length is not a non-negative integer")
+                if stream:
+                    stream.expected_content_length = content_length
+            elif key == b"transfer-encoding" and value != "trailers":
```

Comment:
Shouldn't this be `b"trailers"` ? I'm not sure we have a test for the happy-path here :)